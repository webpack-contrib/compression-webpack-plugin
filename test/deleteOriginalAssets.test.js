import Plugin from '../src/index';

import {
  cleanErrorStack,
  createCompiler,
  compile,
  getAssetsInfo,
} from './helpers';

describe('when applied with `test` option', () => {
  let compiler;

  beforeEach(() => {
    compiler = createCompiler({
      entry: {
        js: `${__dirname}/fixtures/entry.js`,
      },
    });
  });

  it('matches snapshot for `true` value ({Boolean})', () => {
    new Plugin({
      minRatio: 1,
      deleteOriginalAssets: true,
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });

  it('matches snapshot for `false` value ({Boolean})', () => {
    new Plugin({
      minRatio: 1,
      deleteOriginalAssets: false,
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });
});
