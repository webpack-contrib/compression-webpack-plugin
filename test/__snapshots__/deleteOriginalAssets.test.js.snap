// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`test\` option matches snapshot for \`false\` value ({Boolean}): assets 1`] = `
Array [
  Array [
    "1.async.9ada34e1a37e6db8ef40.js",
    130,
  ],
  Array [
    "1.async.9ada34e1a37e6db8ef40.js.gz",
    117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.f0111e97a13538df48f2.js",
    10212,
  ],
  Array [
    "js.f0111e97a13538df48f2.js.gz",
    2769,
  ],
]
`;

exports[`when applied with \`test\` option matches snapshot for \`false\` value ({Boolean}): errors 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot for \`false\` value ({Boolean}): warnings 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot for \`true\` value ({Boolean}): assets 1`] = `
Array [
  Array [
    "1.async.9ada34e1a37e6db8ef40.js.gz",
    117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.f0111e97a13538df48f2.js.gz",
    2769,
  ],
]
`;

exports[`when applied with \`test\` option matches snapshot for \`true\` value ({Boolean}): errors 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot for \`true\` value ({Boolean}): warnings 1`] = `Array []`;
